fix: Explicitly mark parameters as nullable

dimitri-guitton opened the pull request:
This pull request addresses the deprecation warning related to implicitly marking parameters as nullable.

dimitri-guitton committed:
fix: Implicitly marking parameter as nullable is deprecated

dimitri-guitton committed:
fix(tests): Implicitly marking parameter as nullable is deprecated, the explicit nullable type must be used instead

nikolaposa commented on src/Version.php:
```diff
@@ -31,7 +31,8 @@ class Version implements JsonSerializable
 
-    final protected function __construct(int $major, int $minor, int $patch, PreRelease $preRelease = null, Build $build = null)
+
+    final protected function __construct(int $major, int $minor, int $patch, ?PreRelease $preRelease = NULL, ?Build $build = NULL)
```
```suggestion
    final protected function __construct(int $major, int $minor, int $patch, ?PreRelease $preRelease = null, ?Build $build = null)
```

nikolaposa commented on src/Version.php:
```diff
@@ -44,7 +45,8 @@ final protected function __construct(int $major, int $minor, int $patch, PreRele
 
-    public static function from(int $major, int $minor = 0, int $patch = 0, PreRelease $preRelease = null, Build $build = null): Version
+
+    public static function from(int $major, int $minor = 0, int $patch = 0, ?PreRelease $preRelease = NULL, ?Build $build = NULL): Version
```
```suggestion
    public static function from(int $major, int $minor = 0, int $patch = 0, ?PreRelease $preRelease = null, ?Build $build = null): Version
```

nikolaposa commented on tests/TestAsset/VersionIsIdentical.php:
```diff
@@ -1,55 +1,61 @@
+            int $expectedMajor,
+            int $expectedMinor,
+            int $expectedPatch,
+            PreRelease|string|null $expectedPreRelease = NULL,
+            Build|string|null $expectedBuild = NULL
```
```suggestion
            PreRelease|string|null $expectedPreRelease = null,
            Build|string|null $expectedBuild = null
```

nikolaposa commented on tests/TestAsset/VersionIsIdentical.php:
```diff
@@ -1,55 +1,61 @@
+            );
+        }
+
+
```
```suggestion
```

nikolaposa commented on tests/TestAsset/VersionIsIdentical.php:
```diff
@@ -1,55 +1,61 @@
+            );
+        }
+
+
```
```suggestion
```

nikolaposa commented on tests/TestAsset/VersionIsIdentical.php:
```diff
@@ -1,55 +1,61 @@
+                    || (!$version->isPreRelease() && !$this->expectedVersion->isPreRelease())
+                )
+                && (
+                    ($version->hasBuild() && $this->expectedVersion->hasBuild() && $version->getBuild()->getIdentifiers() === $this->expectedVersion->getBuild()
+                                                                                                                                                    ->getIdentifiers(
+                                                                                                                                                    ))
+                    || (!$version->hasBuild() && !$this->expectedVersion->hasBuild())
```
Please don't make style changes like these, it doesn't help readability at all. If you really wanted to improve something, you could have done something like:
```suggestion
                    (
                        $version->hasBuild() 
                        && $this->expectedVersion->hasBuild() 
                        && $version->getBuild()->getIdentifiers() === $this->expectedVersion->getBuild()->getIdentifiers()
                    ) || (!$version->hasBuild() && !$this->expectedVersion->hasBuild())
```

dimitri-guitton committed:
refactor : update code to comply with repository standards

dimitri-guitton committed:
refactor : update code to comply with repository standards

dimitri-guitton committed:
refactor : update code to comply with repository standards

nikolaposa commented:
@d-loki Something is very off with code style in your fork. Check out the differences between:

https://github.com/nikolaposa/version/blob/master/src/Version.php

and

https://github.com/nikolaposa/version/blob/bf7c73d71a5aafcad2d3c4801bfae50cc4103a91/src/Version.php

where does that extra tab before every line comes from in your `Version.php`?

![Screenshot 4](https://github.com/user-attachments/assets/848eff8c-f43b-4d4f-a6f6-bb3a88d57033)

dimitri-guitton committed:
fix(PhpStan): Remove deprecated config checkMissingIterableValueType and checkGenericClassInNonGenericObjectType

Replace with :
ignoreErrors:
            - identifier: missingType.generics
            - identifier: missingType.iterableValue

dimitri-guitton committed:
fix(PhpStan): Resolve PHPStan warning for non-nullable 'prefix' offset in array

dimitri-guitton commented:
@nikolaposa I’ll check and fix the issue.
It's a configuration in my IDE. I'll start from your files and only make the necessary changes without letting my IDE update the code.

dimitri-guitton committed:
refactor : update code to comply with repository standards

dimitri-guitton commented:
@nikolaposa I think it should be much better now.
Sorry for these issues, I didn’t notice that my IDE had updated the files during the commit.

nikolaposa commented:
@d-loki Please fix these as well: https://github.com/nikolaposa/version/actions/runs/12990595086/job/36226775577?pr=42

dimitri-guitton committed:
fix : Add newline at end of file VersionIsIdentical

dimitri-guitton committed:
fix :  Add newline at end of file Version

dimitri-guitton committed:
chore : Run php cs fixer

dimitri-guitton commented:
> @d-loki Please fix these as well: https://github.com/nikolaposa/version/actions/runs/12990595086/job/36226775577?pr=42

It's fixed.

VincentLanglet commented:
Solved by https://github.com/nikolaposa/version/pull/43
